app/models/order.rb

```diff
@@ -0,0 +1,15 @@
+class Order < ApplicationRecord
+    belongs_to :user
+    has_many :order_items
+    before_save :set_subtotal
+    
+    def subtotal
+        order_items.collect{|order_item| order_item.valid? ? order_item.unit_price*order_item.quantity : 0}.sum
+    end
```

Comment:
Dont override rails provided methods. As rails automatically adds method for columns in model instance.

Reply:
I need to override it because i have to calculate the price of each order_item inside the order to calculate the total bill.

Reply:
This will calculate the sum whenever called instead of getting the value saved in DB. instead you can move this logic into `set_subtotal` which computes and save the total `self.subtotal = logic` so that you can use `subtotal` function without overriding it.